pkg/controller/core/workload_controller.go

```diff
@@ -85,25 +85,25 @@ func (r *WorkloadReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 	switch status {
 	case pending:
 		if !r.queues.QueueForWorkloadExists(&wl) {
-			err := workload.UpdateStatusIfChanged(ctx, r.client, &wl, kueue.WorkloadAdmitted, corev1.ConditionFalse,
+			err := workload.UpdateStatusIfChanged(ctx, r.client, &wl, kueue.WorkloadAdmitted, metav1.ConditionFalse,
 				"Inadmissible", fmt.Sprintf("Queue %s doesn't exist", wl.Spec.QueueName))
 			return ctrl.Result{}, client.IgnoreNotFound(err)
 		}
 
 		cqName, cqOk := r.queues.ClusterQueueForWorkload(&wl)
 		if !cqOk {
-			err := workload.UpdateStatusIfChanged(ctx, r.client, &wl, kueue.WorkloadAdmitted, corev1.ConditionFalse,
+			err := workload.UpdateStatusIfChanged(ctx, r.client, &wl, kueue.WorkloadAdmitted, metav1.ConditionFalse,
 				"Inadmissible", fmt.Sprintf("ClusterQueue %s doesn't exist", cqName))
 			return ctrl.Result{}, client.IgnoreNotFound(err)
 		}
 
 		if !r.cache.ClusterQueueActive(cqName) {
-			err := workload.UpdateStatusIfChanged(ctx, r.client, &wl, kueue.WorkloadAdmitted, corev1.ConditionFalse,
+			err := workload.UpdateStatusIfChanged(ctx, r.client, &wl, kueue.WorkloadAdmitted, metav1.ConditionFalse,
 				"Inadmissible", fmt.Sprintf("ClusterQueue %s is inactive", cqName))
 			return ctrl.Result{}, client.IgnoreNotFound(err)
 		}
 	case admitted:
-		err := workload.UpdateStatusIfChanged(ctx, r.client, &wl, kueue.WorkloadAdmitted, corev1.ConditionTrue, "", "")
+		err := workload.UpdateStatusIfChanged(ctx, r.client, &wl, kueue.WorkloadAdmitted, metav1.ConditionTrue, "ByClusterQueue", "Admitted by ClusterQueue")
```

Comment:
add the cluster queue in the message? also, for reason, may be we should have a name for the controller and use that instead of "ByClusterQueue"?

Reply:
I thought about adding the name of the ClusterQueue, but I found it repetitive since it's already present in the admission object. Still, I could add it.

For reason, are you thinking along the lines of `ByKueueAdmission`?

Reply:
> I thought about adding the name of the ClusterQueue, but I found it repetitive since it's already present in the admission object. Still, I could add it.

I would add it unless you are concerned about the message size.

> For reason, are you thinking along the lines of ByKueueAdmission?

Whatever string we put here I guess doesn't matter much, but the bigger question is if we will ever have different values for this field? for example if the admission was done manually (the case where admins force start a job)? or in this case the condition will not exist and be empty?

Reply:
I think it would be nice to mark it as admitted by the admin. Although the current code would override it.

In any case, `ByKueueAdmission` would fit well if we ever did that.

Reply:
ok, lets change it then.

Reply:
I'm not a native speaker, but `KueueAdmitted` or `SchedulerAdmitted` seems better to me than `ByKueueAdmission` or `ByXxx`. Forget it if not.

Reply:
We can use this as a reference: https://github.com/kubernetes/website/pull/36007/files
It has the action and an actor, so I'll go with `AdmissionByKueue`